lib/rust/ensogl/core/src/system/gpu/shader/compiler.rs

```diff
@@ -259,12 +264,36 @@ impl CompilerData {
         profiler: profiler::Debug,
         on_ready: F,
     ) -> JobHandler {
-        self.dirty = true;
         let strong_handler = JobHandler::new();
         let handler = strong_handler.downgrade();
         let on_ready = Box::new(on_ready);
-        let job = Job { input, handler, on_ready, profiler };
-        self.jobs.compile.push(job);
+        let cache_key = ShaderCache::code_key(&input);
+
+        // Perform cache lookup before spawning compilation job.
+        match self.cache.get_program(cache_key) {
+            Some(ProgramCacheEntry::Submitted | ProgramCacheEntry::Validated(_)) => {
+                debug!("Reuse cached shader {cache_key:?}.");
```

Comment:
What does this debug message mean? Does it tell the user to reuse the cached shader? Or does it inform the user that the cached shader will be reused? If the latter, it should be written as "Reusing ...".